walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,123 +15,80 @@
-
-    // stores the index in the array from which this meeting begins
-    int startIndex = 0;
-    // stores the time at which this meeting begins
-    int startTime = 0;
-    // stores the index in the array at which this meeting ends
-    int endIndex = 0;
-    // stores the time at which this meeting ends
-    int endTime = 0;
-
-    if (prev < 0) {
-      // If the meeting is to be inserted at the start of the list, startIndex = 0
-      startIndex = startPos;
-      startTime = meetingTime.start();
-    } else {
-      TimeRange prevElem = busyTimes.get(prev);
-      /*
-       * Otherwise find out whether this meeting overlaps with the previous meeting
-       * If it does, coalesce it with the previous meeting otherwise keep them
-       * separate
-       */
-      startIndex = prevElem.overlaps(meetingTime) ? prev : startPos;
-      startTime = prevElem.overlaps(meetingTime) ? prevElem.start() : meetingTime.start();
-    }
-
-    if (next >= busyTimes.size()) {
-      // If this element is to be inserted at the end of the list, endIndex = last element of list
-      endIndex = next - 1;
-      endTime = meetingTime.end();
-    } else {
-      /*
-       * Otherwise find out whether this meeting overlaps with the next meeting
-       * If it does, coalesce it with the next meeting otherwise keep them
-       * separate
-       */
-      TimeRange nextElem = busyTimes.get(next);
-      endIndex = nextElem.overlaps(meetingTime) ? next : next - 1;
-      endTime = nextElem.overlaps(meetingTime) ? nextElem.end() : meetingTime.end();
-    }
+  private int createCombinedBlock(
+      TimeRange block, TreeSet<TimeRange> busyTimes, long duration, HashSet<String> blockOptBusy) {
+    // Stores all time ranges later than this range
+    Iterator<TimeRange> sub = busyTimes.tailSet(block, false).iterator();
+    // Stores the start time of the larger time block
+    int blockStart = block.start();
+    // Stores the end time of the larger time block
+    int blockEnd = block.end();
+    // Stores the duration of the larger time block
+    int blockDuration = block.duration();
 
-    // create a new meeting with the given start and end times
-    TimeRange newBusy = TimeRange.fromStartEnd(startTime, endTime, false);
-
-    /*
-     * This meeting ends before the current first meeting starts, so add it to the
-     * beginning of the list
-     */
-    if (endIndex < 0) {
-      newBusyTimes.add(newBusy);
-    }
-    for (int i = 0; i < busyTimes.size(); i++) {
-      // Insert the new entry in its appropriate location ordered by start times
-      if (i == startIndex && endIndex >= 0) {
-        newBusyTimes.add(newBusy);
-      }
-      // And remove all redundant entries between its start and end points
-      if (i < startIndex || i > endIndex) {
-        newBusyTimes.add(busyTimes.get(i));
+    blockOptBusy.addAll(block.getOptBusy());
+    while (sub.hasNext() && blockDuration < duration) {
```

Comment:
It seems that you will stop as soon as you find a blockDuration that is >= duration. 
What if the event goes like 
`|___req___||-opt-||-opt-||-opt-||-opt-||-opt-||-opt-||-opt-||___req___|` with a duration that is the length of 2 opt blocks? 
Please also add a test case that demonstrates this.